src/backend/elfobj.c

```diff
@@ -1653,17 +1663,18 @@ void obj_tlssections()
     IDXSTR namidx;
     int align = I64 ? 16 : 4;
 
-    int sec = ElfObj::getsegment(".tdata", NULL, SHT_PROGBITS, SHF_ALLOC|SHF_WRITE|SHF_TLS, align);
+    int sec = ElfObj::getsegment(".tdata", NULL, SHT_PROGBITS, tls_section_flags(), align);
     Obj::bytes(sec, 0, align, NULL);
 
     namidx = Obj::addstr(symtab_strings,"_tlsstart");
-    elf_addsym(namidx, 0, align, STT_TLS, STB_GLOBAL, MAP_SEG2SECIDX(sec));
+    const int tls_symbol = (config.flags3 & CFG3android) ? STT_OBJECT : STT_TLS;
+    elf_addsym(namidx, 0, align, tls_symbol, STB_GLOBAL, MAP_SEG2SECIDX(sec));
 
-    ElfObj::getsegment(".tdata.", NULL, SHT_PROGBITS, SHF_ALLOC|SHF_WRITE|SHF_TLS, align);
+    ElfObj::getsegment(".tdata.", NULL, SHT_PROGBITS, tls_section_flags(), align);
 
-    sec = ElfObj::getsegment(".tcommon", NULL, SHT_NOBITS, SHF_ALLOC|SHF_WRITE|SHF_TLS, align);
+    sec = ElfObj::getsegment(".tcommon", NULL, SHT_NOBITS, tls_section_flags(), align);
     namidx = Obj::addstr(symtab_strings,"_tlsend");
-    elf_addsym(namidx, 0, align, STT_TLS, STB_GLOBAL, MAP_SEG2SECIDX(sec));
+    elf_addsym(namidx, 0, align, tls_symbol, STB_GLOBAL, MAP_SEG2SECIDX(sec));
```

Comment:
There is TLS support in C and C++ now, so they'll support it at some point.
Not sure why you're even arguing about this.